### PR TITLE
Docs: use new ActorSystem API

### PR DESCRIPTION
**Pull request by ennru**
More to https://github.com/akka/akka-http/issues/3073

Requires merge of https://github.com/akka/akka-http/pull/3223

Discovered https://github.com/akka/akka-http/issues/3240 and https://github.com/akka/akka-http/issues/3241

### CODE REVIEW

File: docs/src/test/java/docs/http/javadsl/ClientSingleRequestExample.java [hunk 1]
```diff
@@ -45,10 +46,11 @@ public void request() {
   }
 
   public void response() {
-    Materializer materializer = null;
+    ActorSystem<Void> system = null;
     HttpResponse response = null;
+    // TODO https://github.com/akka/akka-http/issues/3240
     //#unmarshal-response-body
-    CompletionStage<Pet> pet = Jackson.unmarshaller(Pet.class).unmarshal(response.entity(), materializer);
+    CompletionStage<Pet> pet = Jackson.unmarshaller(Pet.class).unmarshal(response.entity(), SystemMaterializer.get(system).materializer());
```
Comment (jrudolph) [inline docs/src/test/java/docs/http/javadsl/ClientSingleRequestExample.java:53]: Hopefully that can improved with #3240.

File: docs/src/main/paradox/introduction.md [hunk 2]
```diff
@@ -44,14 +44,16 @@ Alternatively, you can bootstrap a new sbt project with Akka HTTP already
 configured using the [Giter8](http://www.foundweekends.org/giter8/) template:
 
 @@@ div { .group-scala }
-```sh
-sbt -Dsbt.version=1.2.8 new https://github.com/akka/akka-http-quickstart-scala.g8
-```
+For Scala (sbt)
+:  ```sh
+    sbt -Dsbt.version=1.3.12 new akka/akka-http-quickstart-scala.g8
```
Comment (raboof) [inline docs/src/main/paradox/introduction.md:49]: do we still need the `-Dsbt.version`? that only determines the sbt version that `sbt new` is using, not that the generated project is using.

File: docs/src/test/java/docs/http/javadsl/HttpServerActorInteractionExample.java [hunk 3]
```diff
@@ -71,63 +71,74 @@ private Route createRoute() {
...
+    private Behavior<Message> onBid(Bid bid) {
+      bids.add(bid);
+      getContext().getLog().info("Bid complete: {}, {}", bid.userId, bid.offer);
+      return this;
+    }
+
+    private Behavior<Message> onGetBids(GetBids getBids) {
+      getBids.replyTo.tell(new Bids(bids));
+      return this;
+    }
```
Comment (raboof) [inline docs/src/test/java/docs/http/javadsl/HttpServerActorInteractionExample.java:141]: Neat!

File: docs/src/test/scala/docs/http/scaladsl/HttpServerWithActorInteraction.scala [hunk 4]
```diff
@@ -4,51 +4,62 @@
...
-    val auction = system.actorOf(Props[Auction], "auction")
+    val auction: ActorRef[Auction.Message] = system
+    import Auction._
 
     val route =
       path("auction") {
         concat(
           put {
-            parameter("bid".as[Int], "user") { (bid, user) =>
+            parameters(("bid".as[Int], "user")) { (bid, user) =>
```
Comment (raboof) [inline docs/src/test/scala/docs/http/scaladsl/HttpServerWithActorInteraction.scala:62]:
```suggestion
            // TODO https://github.com/akka/akka-http/pull/2971
            parameters(("bid".as[Int], "user")) { (bid, user) =>
```

**Pull request merged**